pkg/api/types.go

```diff
@@ -173,11 +173,116 @@ type VolumeSource struct {
 	HostPath *HostPath `json:"hostPath"`
 	// EmptyDir represents a temporary directory that shares a pod's lifetime.
 	EmptyDir *EmptyDir `json:"emptyDir"`
+	// GitRepo represents a git repository at a particular revision.
+	GitRepo *GitRepo `json:"gitRepo"`
+
 	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
 	GCEPersistentDisk *GCEPersistentDisk `json:"persistentDisk"`
-	// GitRepo represents a git repository at a particular revision.
-	GitRepo *GitRepo `json:"gitRepo"`
+
+	AWSElasticBlockStore *AWSElasticBlockStore `json:"elasticBlockStore"`
+
+	NFSMount *NFSMount `json:"nfsMount"`
+
+	PersistentVolumeClaim *PersistentVolumeClaimAttachment `json:"persistentVolumeClaim,omitempty"`
```

Comment:
@bgrant0607 @thockin   If the desire is to remove these entirely from the user API, does it make sense to implement the claim attachment as a volume?  It seems like this belongs up one level, on Volume.    Let volume have Name and a ClaimAttachment?  We could then refactor VolumeSource out of Volume.

I'll implement this as a volume plugin if that's what you prefer, but wouldn't mind some guidance on how to remove volumes from the user API (leaving it to the admins who still need to define the pets in the system).  Removing that from the API would be in another pull, but I've already started thinking about how it might be implemented.